### PR TITLE
Fix sometimes missing the matching elements with `stopOnDomReady:true

### LINKED ISSUE #27: Sometimes it can't capture the matching elements with `stopOnDomReady=true`
Demo: https://riophae.github.io/element-ready-bug-reproduction/
You can find the source code at https://github.com/riophae/element-ready-bug-reproduction

image

You see `elementReady('#target')` can't find the `div#target` element that it should do. This is repeatable on Chrome 76, Firefox 68 & Edge 18. The OS is Windows 10 x64 1903.
I have also included the fixed version for comparison.

The reason behind this is, the `div#target` element is placed near `</html>`. Before `element-ready` spots it, `dom-loaded` has cancelled `requestAnimationFrame`.

### PR DESCRIPTION
**Pull request by riophae**
Fixes #27

### CODE REVIEW

**Comment by fregante** [comment]
Thank you for the PR, but the fix should be much simpler. Our code stops checking exactly when the DOM becomes ready, but instead when that happens it should do one last query. That’s the only change required

**Comment by riophae** [comment]
Hi @fregante, my fix used to be simple but... It didn't pass the tests. That's why it has changed a lot then. You see there are two commits.

**Comment by riophae** [comment]
OK. I know how to simplify the code now. I'll push one more commit.

File: test.js [hunk 1]
```diff
@@ -8,7 +8,6 @@ global.document = window.document;
 global.requestAnimationFrame = fn => setTimeout(fn, 16);
 global.cancelAnimationFrame = id => clearTimeout(id);
 
-// `dom-loaded` immediately uses `global.document`, so it needs to be called after its definition
 const elementReady = require('.');
```
Comment (fregante) [inline test.js]: This can be reverted to an `import` since dom-loaded has been dropped

**Comment by fregante** [comment]
Cool, that works too. 

I was looking at your previous code and I saw it also fixed a bug we missed before: if the element is found right away, `stop` is called **before** we call `cache.set`, so the promise remains cached even after the resolution. 

In your latest code we could probably also fix this issue with something like:

```js
		stop = async () => {
			cancelAnimationFrame(rafId);
			resolve();
			await promise;
			cache.delete(cacheKeys, promise);
		};
``` 

Also needs a comment to explain why it’s there

**Comment by riophae** [comment]
Well, with `p-defer`, we can cache the promise before the first calling of `stop()`. Then we don't need to fix it at all :)

**Comment by fregante** [comment]
D’oh, indeed. Let’s revert my last commit then 🤦‍♂️

**Comment by riophae** [comment]
I just updated the demo with the new fix, and you see it's working well: https://riophae.github.io/element-ready-bug-reproduction/

**Comment by riophae** [comment]
I just added some comments... Then I realize the previous version without `dom-loaded` would be better. Because codes that don't need comments are better codes :)

**Comment by fregante** [comment]
Thanks @riophae, good stuff 💯💯

**Pull request merged**